kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java

```diff
@@ -333,19 +350,56 @@ public static boolean isValidName(String name) {
     return Utils.isNotNullOrEmpty(name) && KUBERNETES_DNS1123_LABEL_REGEX.matcher(name).matches();
   }
 
+  /**
+   * Validates annotation or label key.
+   *
+   * @param key
+   * @return returns a boolean value indicating whether it's valid or not
+   */
+  public static boolean isValidKey(String key) {
+    return Utils.isNotNullOrEmpty(key) && key.length() < 254 && KUBERNETES_KEY_REGEX.matcher(key).matches();
+  }
+
+  /**
+   * Validates name of an resource according to DNS Subdomain rules.
+   * <p>
+   * Refer to <a href="https://kubernetes.io/docs/concepts/overview/working-with-objects/names/">Kubernetes Naming
+   * Conventions</a>
+   *
+   * @param name Name of annotation
+   * @return returns a boolean value indicating whether it's valid or not
+   */
+  public static boolean isValidSubdomainName(String name) {
+    return Utils.isNotNullOrEmpty(name) && name.length() < 254 && KUBERNETES_SUBDOMAIN_REGEX.matcher(name).matches();
+  }
+
   /**
    * Validates labels/annotations of Kubernetes resources
    *
    * @param map Label/Annotation of resource
    * @return returns a boolean value indicating whether it's valid or not
+   *
+   * @see #areLabelsValid(Map) {@link #areAnnotationsValid(Map)}
+   *
+   * @deprecated the rules are different for label and annotation values
    */
+  @Deprecated
   public static boolean isValidLabelOrAnnotation(Map<String, String> map) {
-    for (Map.Entry<String, String> entry : map.entrySet()) {
-      if (!(isValidName(entry.getKey()) && isValidName(entry.getValue()))) {
-        return false;
-      }
-    }
-    return true;
+    return areLabelsValid(map);
+  }
+
+  /**
+   * Checks the given map keys and values for validity as labels.
+   */
+  public static boolean areLabelsValid(Map<String, String> map) {
```

Comment:
javadoc seems to be missing `@param` and `@return`

Reply:
I got a little lazy as they seemed self-explanitory.  I'll add something.